multiqc/modules/lima/lima.py

```diff
@@ -0,0 +1,365 @@
+            lima_counts[sample] = {"Counts": int(counts), "MeanScore": float(mean_score)}
+
+        return lima_counts
+
+    def update_sample_names(self, data):
+        lima_barcodes = config.kwargs["lima_barcodes"]
+        # If --lima-barcodes wasn't specified
+        if not lima_barcodes:
+            return
+
+        # Read the barcodes to sample mapping
+        barcodes = self.read_lima_barcodes(lima_barcodes)
+
+        # Copy the data over to the new dictionary by sample name
+        for name in list(data.keys()):
+            # If we have a mapping from the barcode to samplename
+            if name in barcodes:
+                # Get the new sample name
+                sample_name = barcodes[name]
+                # Update the sample name
+                data[sample_name] = data.pop(name)
+
+    def read_lima_barcodes(self, lima_barcodes):
+        """
+        Read the lima barcodes file, and return as a dictionary
+
+        The keys will be of the format barcode1--barcode2, and the values will
+        be the sample name
+        """
+        barcodes = dict()
+        with open(lima_barcodes) as fin:
+            for line in fin:
+                sample, first_barcode, second_barcode = line.strip().split("\t")
+                barcodes[f"{first_barcode}--{second_barcode}"] = sample
+        return barcodes
+
+    def add_sections(self):
+        plot_data = dict()
+
+        # First, we gather all filter results for each lima summary
+        all_filters = set()
+        for filename, data in self.lima_summary.items():
+            for reason in self.filter_and_pass(data):
+                all_filters.add(reason)
+
+        plot_data = list()
+        for filename, data in self.lima_summary.items():
+            d = dict()
+            for reason in all_filters:
+                d[reason] = dict()
+                # This is where the filter reasons are stored
+                zmw_marginals = data["ZMWs below any threshold"]["ZMW marginals"]
+                for filter_reason in zmw_marginals:
+                    if filter_reason == reason:
+                        d[reason][filename] = zmw_marginals[filter_reason]["count"]
+                        break
+                # If we didn't find filter reason for this filename, set it to
+                # zero
+                else:
+                    d[reason][filename] = 0
+                # Unless 'reason' is actually the special case of the number of
+                # unfiltered reads
+                if reason == "ZMWs above all thresholds":
+                    d[reason][filename] = data["ZMWs above all thresholds"]["count"]
+            plot_data.append(d)
+
+        # Plot configuration
+        config = {
+            "id": "lima-filter-graph",
+            "title": "Lima: ZMW results",
+            "ylab": "Number of ZMWs",
+            "xlab": "Lima summary file",
+            "cpswitch": False,
+            "logswitch": True,
+            "stacking": None,
```

Comment:
I generally say that we should not go for this approach - no stacking and using tabs for samples.

Firstly - it doesn't scale well. If there are samples with long names, or if there are a large number of samples then the report quickly becomes unusable (I try to plan every module as if someone will run 1000 samples).

Secondly - the main purpose of MultiQC is to allow people to get a fast summary of their project and spot any outlier samples that may need closer inspection. By using tabs, the end user has to click through every tab to see every sample and we lose this overview.

As a result, stacked bars showing all samples at once is the preferable display.